src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -114,33 +113,19 @@ public void clusterChanged(ClusterChangedEvent event) {
                 }
             }
 
-            // check and write changes in indices
-            for (IndexMetaData indexMetaData : newMetaData) {
-                String writeReason = null;
-                IndexMetaData currentIndexMetaData;
-                if (currentMetaData == null) {
-                    // a new event..., check from the state stored
-                    try {
-                        currentIndexMetaData = metaStateService.loadIndexState(indexMetaData.index());
```

Comment:
Could we skip this loading from disk? I made the change so that we still load (see `maybeLoadIndexState()` below) if the meta data is not in memory but fail to understand why we do it. Is it an optimization?